src/hotspot/share/nmt/memMapPrinter.cpp

```diff
@@ -127,11 +127,21 @@ class CachedNMTInformation : public VirtualMemoryWalker {
   // Given a vma [from, to), find all regions that intersect with this vma and
   // return their collective flags.
   MemFlagBitmap lookup(const void* from, const void* to) const {
+    assert(from <= to, "Sanity");
+    // We optimize for sequential lookups. Since this class is used when a list
+    // of OS mappings is scanned (VirtualQuery, /proc/pid/maps), and these lists
+    // are usually sorted in order of addresses, ascending.
+    static uintx last = 0;
```

Comment:
Sorry, I should've caught this when we first reviewed this, but why are we using `uintx` instead of `uintptr_t`?

Reply:
Technically these are the same. I tend to prefer `uintx` when its indexes, not addresses, they designate. As in "things you count" vs "things you point to".

Reply:
In other parts of HotSpot we have been changed usages of `uintx` to instead use `size_t`.

Reply:
> In other parts of HotSpot we have been changed usages of `uintx` to instead use `size_t`.

Why? size_t implies memory size to me. This is not a memory size, its a counter.

(If reviewers prefer uintx, I can change it; I just want to understand the reasoning behind choosing size_t)

Reply:
Why `uintx` rather than just `uint`? Is there a reason to have a variable-sized counter for this? Does it even need to be unsigned?

Reply:
Hmm... I guess that would work.

Its number of NMT ranges. 

My original thought was: we have a 64-bit address space, usually 128TB, which would give us 32 billion 4K pages, and with region folding (since you need regions separated by a page) this comes to max. 16 billion 4K regions mapped, which would be more than a 32-bit value can represent.

OTOH maybe that's overthinking. unsigned gives us 4 billion max, and the largest processes I have ever seen had  ~20 million mappings. I think a lot of other things start failing before we reach 4 billion.

Okay, I make it an unsigned then.

Reply:
> > In other parts of HotSpot we have been changed usages of `uintx` to instead use `size_t`.
> 
> Why? size_t implies memory size to me. This is not a memory size, its a counter.
> 
> (If reviewers prefer uintx, I can change it; I just want to understand the reasoning behind choosing size_t)

I don't think it necessarily implies memory sizes. Take a look at the description here:

https://en.cppreference.com/w/cpp/types/size_t

> std::size_t is commonly used for array indexing and loop counting. Programs that use other types, such as unsigned int, for array indexing may fail on, e.g. 64-bit systems when the index exceeds [UINT_MAX](https://en.cppreference.com/w/cpp/types/climits) or if it relies on 32-bit modular arithmetic.